src/Molecules/Duplicable/Duplicable.jsx

```diff
@@ -0,0 +1,60 @@
+import React, { useState } from 'react'
+
+const computeNextKey = duplicated => {
+  let nextKey = 0
+
+  duplicated.map(duplicate => {
+    const key = parseInt(duplicate.key)
+
+    if ('key' in duplicate && key >= nextKey) {
+      nextKey = key + 1
+    }
+  })
+
+  return nextKey
+}
+
+const Duplicable = ({
+  model,
+  addType,
+  canBeEmpty = false,
+  instancesProps = [],
+}) => {
+  const [duplicated, setDuplicated] = useState(
+    instancesProps.length ? instancesProps.map((instanceProps, index) => {
+      return React.cloneElement(model, { key: index, ...instanceProps, index: index })
+    }) :
+    canBeEmpty ? [] :
+    [React.cloneElement(model, { key: 0, index: 0 })],
+  )
```

Comment:
we could factorize this piece of code

```suggestion
  if (instancesProps.length === 0 && !canBeEmpty) {
    instancesProps.push({})
  }
  
  const [duplicated, setDuplicated] = useState(
    instancesProps.map((instanceProps, index) => React.cloneElement(model, { ...instanceProps, key: index, index }))
  )
```